version/version.go

```diff
@@ -55,22 +86,34 @@ type PluginVersion struct {
 	// then it means that it is a final release. Otherwise, this is a pre-release
 	// such as "dev" (in development), "beta", "rc1", etc.
 	versionPrerelease string
+	// Extra metadata that can be part of the version.
+	//
+	// This is legal in semver, and has to be the last part of the version
+	// string, starting with a `+`.
+	versionMetadata string
 	// The Semantic Version of the plugin. Used for version constraint comparisons
 	semVer *version.Version
 }
 
-func (p *PluginVersion) FormattedVersion() string {
-	var versionString bytes.Buffer
-	fmt.Fprintf(&versionString, "%s", p.version)
-	if p.versionPrerelease != "" {
-		fmt.Fprintf(&versionString, "-%s", p.versionPrerelease)
+func (p *PluginVersion) SetMetadata(meta string) {
+	p.versionMetadata = meta
+}
 
-		if GitCommit != "" {
-			fmt.Fprintf(&versionString, " (%s)", GitCommit)
+func (p *PluginVersion) FormattedVersion() string {
```

Comment:
Could this be a breaking change?

Reply:
I don't know if I should call that a "breaking" change, but there's some potential extra information here, normally if a plugin creates their version with core + pre-release, the format should not change at all.
If there's a metadata however, this could change (metadata right now could be defined as `GitCommit` right now already, but it's global)

Reply:
My concern was if Packer was using this same method and showing `git commit` between parenthesis it could be a breaking change for users checking git commits of builds, but Packer doesn't do that. Could it be a problem for packer plugin maintainers? I could see this happening if someone is using the plugin describe command for some validation or using the method to write user-agent to HTTP calls 🤔

Reply:
Yeah that's a good callout, I didn't pay attention that I was changing the format honestly, I thought the commit was added as a metadata to the version if defined. For clarity though, it's not automagic, it needs either to be set by the plugin developers, or as a ldflag. AFAICT we don't set it in the scaffolding, or in any of our plugins.
Also, we use `String` for printing out the version of the plugin for `describe`. I'm not certain this is used anywhere, but I cannot certify it.

That said, I'll do a reroll of this commit to avoid changing the commit format, so it remains complete as it was beforehand.
I'll add a test to make sure it doesn't change between the two versions.